firebase/src/core/domain/models/collection.ts

```diff
@@ -84,12 +84,12 @@ export const localCollectionSchema = publicCollectionSchema.append({
   memos: Joi.array().items(memoValidationSchema).unique().min(1).required(),
 });
 
-export const storedCollectionSchema = publicCollectionSchema.append({
-  memosAmount: Joi.number().integer().min(1).required(),
-  memosOrder: Joi.array().items(Joi.string()).min(1).required(),
-});
-// TODO: How to compare between same-object properties?
-// .assert(Joi.ref("memosAmount"), Joi.ref("memosOrder.length"));
+export const storedCollectionSchema = publicCollectionSchema
+  .append({
+    memosAmount: Joi.number().integer().min(1).required(),
+    memosOrder: Joi.array().items(Joi.string()).unique().min(1).required(),
+  })
+  .assert(".memosAmount", Joi.ref("memosOrder.length"));
```

Comment:
Oh that was tricky, really nice. Where did you find the notation for this usage? Maybe we could add a reference to the docs? I tried and didn't find anything.

Reply:
While searching where I found this solution, I found a better one here: https://github.com/sideway/joi/pull/1206.

Let me know what do you think

Reply:
Added docs